go/vt/vtgate/engine/route.go

```diff
@@ -211,55 +151,16 @@ const (
 	SelectNext
 	// ExecDBA is for executing a DBA statement.
 	ExecDBA
-	// UpdateUnsharded is for routing an update statement
-	// to an unsharded keyspace.
-	UpdateUnsharded
-	// UpdateEqual is for routing an update statement
-	// to a single shard: Requires: A Vindex, and
-	// a single Value.
-	UpdateEqual
-	// DeleteUnsharded is for routing a delete statement
-	// to an unsharded keyspace.
-	DeleteUnsharded
-	// DeleteEqual is for routing a delete statement
-	// to a single shard. Requires: A Vindex, a single
-	// Value, and a Subquery, which will be used to
-	// determine if lookup rows need to be deleted.
-	DeleteEqual
-	// DeleteSharded is for routing a scattered
-	// delete statement.
-	DeleteSharded
-	// InsertUnsharded is for routing an insert statement
-	// to an unsharded keyspace.
-	InsertUnsharded
-	// InsertUnsharded is for routing an insert statement
-	// to a single shard. Requires: A list of Values, one
-	// for each ColVindex. If the table has an Autoinc column,
-	// A Generate subplan must be created.
-	InsertSharded
-	// InsertShardedIgnore is for INSERT IGNORE and
-	// INSERT...ON DUPLICATE KEY constructs.
-	InsertShardedIgnore
-	// NumCodes is the total number of opcodes for routes.
-	NumCodes
 )
 
 var routeName = map[RouteOpcode]string{
-	SelectUnsharded:     "SelectUnsharded",
-	SelectEqualUnique:   "SelectEqualUnique",
-	SelectEqual:         "SelectEqual",
-	SelectIN:            "SelectIN",
-	SelectScatter:       "SelectScatter",
-	SelectNext:          "SelectNext",
-	ExecDBA:             "ExecDBA",
-	UpdateUnsharded:     "UpdateUnsharded",
-	UpdateEqual:         "UpdateEqual",
-	DeleteUnsharded:     "DeleteUnsharded",
-	DeleteSharded:       "DeleteSharded",
-	DeleteEqual:         "DeleteEqual",
-	InsertUnsharded:     "InsertUnsharded",
-	InsertSharded:       "InsertSharded",
-	InsertShardedIgnore: "InsertShardedIgnore",
+	SelectUnsharded:   "SelectUnsharded",
+	SelectEqualUnique: "SelectEqualUnique",
+	SelectEqual:       "SelectEqual",
+	SelectIN:          "SelectIN",
+	SelectScatter:     "SelectScatter",
+	SelectNext:        "SelectNext",
+	ExecDBA:           "ExecDBA",
```

Comment:
For consistency and clarity with the rest of this refactor, it seems to me like we should rename `ExecDBA` to `SelectDBA` and rename the `Route` struct to `Select`.

Then `route.go` would keep all the generic routing utilities like `execShard` but the specific handling of the various select opcodes would move into `select.go`.

Reply:
Agree. I'll do this on my next iteration, which is writing tests for ~Route~ Select.